portfolio/src/main/webapp/script.js

```diff
@@ -160,17 +165,71 @@ function deleteComments() {
   fetch(request).then(response => getComments());
 }
 
+/** Fetches login and Blobstore. */
+function fetchHome() {
+  fetchLogin();
+  fetchBlobstoreUrl();
+}
+
 /**
- * Fetch the Blobstore URL and append it to the comments form
+ * Fetches the Blobstore URL and append it to the comments form
  * in the action attribute.
  */
 function fetchBlobstoreUrl() {
   fetch('/blobstore-upload-url')
-      .then((response) => {
-        return response.text();
-      })
-      .then((imageUploadUrl) => {
+      .then(response => response.text())
+      .then(imageUploadUrl => {
         const messageForm = document.getElementById('comments-form');
         messageForm.action = imageUploadUrl;
-      })
+      });
+}
+
+/**
+ * Fetches the Login Servlet and displayes a message
+ * according to the condition of the user being logged in or out.
+ */
+function fetchLogin() {
+  fetch('/login').then(response => response.json()).then(user => {
+    if (user.loggedIn == true) {
+      // Create element that welcomes the user and prompts him to log out.
+      const welcomeMessage = document.createElement('h1');
+      welcomeMessage.innerHTML = 'Hello, ' + user.email + '!';
+
+      const logoutUrl = document.createElement('a');
+      logoutUrl.href = user.logUrl;
+      logoutUrl.innerText = 'here';
+
+      const logoutPrompt = document.createElement('p');
+      logoutPrompt.innerHTML = 'Click ';
+      logoutPrompt.appendChild(logoutUrl);
+      logoutPrompt.innerHTML += ' to log out.';
+
+      const loggedInMessage = document.getElementById('log');
+      loggedInMessage.innerHTML = '';
+      loggedInMessage.appendChild(welcomeMessage);
+      loggedInMessage.appendChild(logoutPrompt);
+
+      // Show comments form.
+      const commentForm = document.getElementById('comments-form');
+      commentForm.classList.remove('hidden');
+    } else {
+      // Create element that prompts the user to login.
+      const loginUrl = document.createElement('a');
+      loginUrl.href = user.logUrl;
+      loginUrl.innerText = 'log in';
+
+      const loginPrompt = document.createElement('p');
+      loginPrompt.innerHTML = 'To leave a comment, ';
+      loginPrompt.appendChild(loginUrl);
+      loginPrompt.innerHTML += '!';
+
+      const loggedOutMessage = document.getElementById('log');
+      loggedOutMessage.innerHTML = '';
+      loggedOutMessage.appendChild(loginPrompt);
+
+      // Show comments form.
```

Comment:
This should say 'hide comments form', right?

Reply:
Yes! Done.